sigma/plugins/fun/xkcd.py

```diff
@@ -12,26 +12,37 @@
 
         comic_no = args[0]
 
-    random_comic = (len(args) == 0)
-        # True if args is empty
-        # False otherwise
+    random_comic = (len(args) == 0) #True if args is empty, False otherwise
 
-    while True:
-        if not random_comic: joke_url = 'http://xkcd.com/' + comic_no + '/info.0.json'
-        else:                joke_url = 'http://xkcd.com/info.0.json'
+    if not random_comic:
+        joke_url = 'http://xkcd.com/' + comic_no + '/info.0.json'
 
         async with aiohttp.ClientSession() as session:
             async with session.get(joke_url) as data:
                 try: joke_json = await data.json()
                 except Exception:
                     await message.channel.send(None, embed=discord.Embed(title=':exclamation: Requested xkcd doesn\'t exist', color=0x993333))
                     return
+    else:
+        joke_url = 'http://xkcd.com/info.0.json'
 
-        if random_comic:
-            comic_no = str(random.randint(1, joke_json['num']))
-            random_comic = False
-            continue
-        break
+        async with aiohttp.ClientSession() as session:
+            async with session.get(joke_url) as data:
+                try: joke_json = await data.json()
+                except Exception:
+                    await message.channel.send(None, embed=discord.Embed(title=':exclamation: Requested xkcd doesn\'t exist', color=0x993333))
+                    return
+        
+        comic_no = str(random.randint(1, joke_json['num']))
+
```

Comment:
useless newline?

Reply:
Blank lines are useful tools to keep everything looking neat. 

[Readability counts.](https://www.python.org/dev/peps/pep-0020/)

Reply:
Yes, but not for separating single line statements. A blank line is ok every 2-3 lines usually but no every other line.

Reply:
Fixed in 241bf57